ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -363,4 +392,47 @@ public void testListTablesStatement() throws Exception {
 
     assertEquals(expectedTable, testTables.get(0));
   }
+
+  @Test
+  public void shouldFailForIncorrectCSASStatementResultType() throws Exception {
+    KsqlResource testResource = TestKsqlResourceUtil.get();
+    String ksqlString1 = "CREATE STREAM s1 AS SELECT * FROM test_table;";
+
+    Response response1 = testResource.handleKsqlStatements(new KsqlRequest(ksqlString1, Collections
+        .EMPTY_MAP));
+    KsqlEntityList result1 = (KsqlEntityList) response1.getEntity();
+    assertTrue("Incorrect response size.", result1.size() == 1);
+    assertThat(result1.get(0), instanceOf(ErrorMessageEntity.class));
+    ErrorMessageEntity errorMessageEntity1 = (ErrorMessageEntity) result1.get(0);
+    assertTrue(errorMessageEntity1.getErrorMessage().getMessage().equalsIgnoreCase("Invalid "
```

Comment:
As i said above, we should use `assertThat` here too

Reply:
Done.